### ISSUE TITLE
DNS resolution failed in DBless Kong

### ISSUE THREAD

**Issue by sahil-sharma** [Kong/kong#7588]
### Summary
- Operating system: Ubuntu20.04
- Kong version (`$ kong version`): v2.4
- Kong Ingress Controller: v1.3
- K8s version: v1.21.3
- CoreDNS: v1.8.0
- A concise summary of the bug:
Not able to hit an outside API end-point via Kong using `externalService`. Getting some DNS server error.

### Steps To Reproduce
I simply followed the httpbin.org mentioned in the official kong [example](https://docs.konghq.com/kubernetes-ingress-controller/1.3.x/guides/using-external-service/).

### Additional Details & Logs
- Kong proxy error logs (`kubectl logs -f ingress-kong-6475fc9b7d-lqhhz -n kong -c proxy`)
```
2021/07/22 12:18:57 [warn] 24#0: *1386 [lua] base.lua:659: queryDns(): [upstream:proxy-to-httpbin.default.80.svc 1] querying dns for httpbin.org failed: dns server error: 2 server failure. Tried ["(short)httpbin.org:(na) - cache-miss","httpbin.org.kong.svc.cluster.local:33
```

- Output of `/etc/resolv.conf` file in Kong-Proxy pod:
```
/ $ cat /etc/resolv.conf 
nameserver 10.96.0.10
search kong.svc.cluster.local svc.cluster.local cluster.local ap-south-1.compute.internal
options ndots:5
```
There is a similar github issues [[5455](https://github.com/Kong/kong/issues/5455), ] on the same error but, I am not able to resolve it. Any hints would be highly appreciated.

**Comment by sahil-sharma** [comment]
When I fire a curl request to Kong I get this: (Kong is exposing via NodePort)
```
curl -i http://<Kong-IP>:<Kong-NodePort>/foo
HTTP/1.1 503 Service Temporarily Unavailable
Date: Thu, 22 Jul 2021 13:11:43 GMT
Content-Type: application/json; charset=utf-8
Connection: keep-alive
Content-Length: 58
X-Kong-Response-Latency: 0
Server: kong/2.4.1

{"message":"failure to get a peer from the ring-balancer"}%                                                                             
```

**Comment by sahil-sharma** [comment]
I noticed one issue not sure whether this is related to this or not. I checked `Ingress-Controller` logs and I noticed this:
```
time="2021-07-22T12:12:54Z" level=error msg="failed to initialize reporter: lookup kong-hf.konghq.com on 10.96.0.10:53: server misbehaving" component=reporter
```

**Comment by sahil-sharma** [comment]
Similar issue on this: [6812](https://github.com/Kong/kong/issues/6812)

**Comment by fffonion** [comment]
@sahil-sharma that feels like an environment problem since you are also not able to resolve address outside of Kong.
Would you be able to run `dig google.com` inside the pod or another pod in the same cluster?

**Comment by sahil-sharma** [comment]
@fffonion Yes, I too feel that it seems to be an env related glitch. When I try on EKS the same works fine. I still need to figure this out why this is not working on local set-up.

**Comment by sahil-sharma** [comment]
@fffonion Surprisingly, I am not able to dig anything outside the cluster. There is no restart in the core-dns Pod or nothing suspicious as I followed this approach (to set up the cluster) umpteen times. Very recently I started facing this issue.
Below are the details:
Env I am having is:
```
OS: 20.04 (vagrant machine)
Kong: 2.4.0
Ingress Controller: 1.3
K8s version: v1.20.1
```
I faced a similar issue when I tried on AWS EC2 instance and installed a K8s cluster with kubeadm. Not sure what wrong I am doing.
Do you feel it would be something related to `ndots` in `/etc/resolv.conf` file?
Any hints.

**Comment by sahil-sharma** [comment]
I dig a bit into the issue and found that there seems to be an issue with `coredns` with `flannel` (CNI) as there were errors in `coredns` logs:
![image](https://user-images.githubusercontent.com/5774379/128628037-05057c33-43cd-4dc5-8b8a-de6208399630.png)
There are some Github issues and Stackoverflow questions on the same error but, I was confused about what to do. Some suggested installing kernel module `modprobe br_netfilter` which was already installed on the system where K8s was running.
Then I looked for an alternate CNI (calico) and installed it and it worked fine.
![image](https://user-images.githubusercontent.com/5774379/128628119-23ac3533-792e-4425-8d30-4c8166ba62f0.png)
I was able to ping DNS outside the cluster DNS.
There must be some other solutions like clearing `iptables` or adding some `FORWARD` or `ACCEPT` rules to it. Or using `dnsPolicy` in deployment but, I prefer to have an easy solution.
Anyhow, I am closing this ticket as I am able to resolve it with some other trick but, the actual problem still persists.
